server/tests/lib/nl/detection/llm_detector_test.py

```diff
@@ -0,0 +1,215 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+from parameterized import parameterized
+
+from server.lib.nl.detection import llm_detector
+from server.lib.nl.detection.types import ClassificationType
+from server.lib.nl.detection.types import ContainedInClassificationAttributes
+from server.lib.nl.detection.types import ContainedInPlaceType
+from server.lib.nl.detection.types import CorrelationClassificationAttributes
+from server.lib.nl.detection.types import EventClassificationAttributes
+from server.lib.nl.detection.types import EventType
+from server.lib.nl.detection.types import NLClassifier
+from server.lib.nl.detection.types import QCmpType
+from server.lib.nl.detection.types import Quantity
+from server.lib.nl.detection.types import QuantityClassificationAttributes
+from server.lib.nl.detection.types import RankingClassificationAttributes
+from server.lib.nl.detection.types import RankingType
+from server.lib.nl.detection.types import SizeType
+from server.lib.nl.detection.types import SizeTypeClassificationAttributes
+from server.lib.nl.detection.types import TimeDeltaClassificationAttributes
+from server.lib.nl.detection.types import TimeDeltaType
+
+
+class TestMergeSV(unittest.TestCase):
+
+  @parameterized.expand([
+      # When there's only 1 SV, we get what we give.
+      (['population'], [{
+          'SV': [
+              'Count_Person', 'GrowthRate_Count_Person', 'Count_Person_Female'
+          ],
+          'CosineScore': [0.9, 0.8, 0.4]
+      }], {
+          'SV': [
+              'Count_Person', 'GrowthRate_Count_Person', 'Count_Person_Female'
+          ],
+          'CosineScore': [0.9, 0.8, 0.4]
+      }),
+      # When there are multiple, we get a MultiSV.
+      (['asthma', 'poverty'], [{
+          'SV': ['Count_Person_Asthma', 'Percent_Person_Asthma'],
+          'CosineScore': [0.9, 0.8]
+      }, {
+          'SV': ['Count_Person_BelowPoverty', 'Count_Person_FoodStamps'],
+          'CosineScore': [0.7, 0.4]
+      }], {
+          'CosineScore': [0.9, 0.8],
```

Comment:
why the CosineScore is only for the first sv here?

Reply:
Good Q.  Because in this case of LLMs detecting multi-SV we *always* set AggCosineScore to be 1.0 so the MultiSV candidate will be picked over SV (single-sv candidate).  So the CosineScore does not matter as long as its <1.0
 https://github.com/datacommonsorg/website/blob/master/server/lib/nl/detection/llm_detector.py#L227-L228.